src/data/polynomial/unit_trinomial.lean

```diff
@@ -0,0 +1,315 @@
+  λ p, ⟨finsupp.filter (set.Ioo (k + n) (n + n)) p.to_finsupp⟩,
+  replace h := congr_arg f h,
+  replace h := (irreducible_aux1 hkm hmn hu hw hp).trans h,
+  replace h := h.trans (irreducible_aux1 hkm' hmn' hu hw hq).symm,
+  rw (is_unit_C.mpr hv).mul_right_inj at h,
+  rw binomial_eq_binomial hu.ne_zero hw.ne_zero at h,
+  simp only [add_left_inj] at h,
+  rcases h with ⟨rfl, -⟩ | ⟨rfl, rfl, h⟩ | ⟨-, hm, hm'⟩,
+  { exact or.inl (hq.trans hp.symm) },
+  { refine or.inr _,
+    rw [←trinomial_mirror hkm' hmn' hu.ne_zero hw.ne_zero, eq_comm, mirror_eq_iff] at hp,
+    exact hq.trans hp },
+  { suffices : m = m',
+    { rw this at hp,
+      exact or.inl (hq.trans hp.symm) },
+    rw [tsub_add_eq_add_tsub hmn.le, eq_tsub_iff_add_eq_of_le, ←two_mul] at hm,
+    rw [tsub_add_eq_add_tsub hmn'.le, eq_tsub_iff_add_eq_of_le, ←two_mul] at hm',
+    exact mul_left_cancel₀ two_ne_zero (hm.trans hm'.symm),
+    exact hmn'.le.trans (nat.le_add_right n k),
+    exact hmn.le.trans (nat.le_add_right n k) },
+end
+
+lemma irreducible_aux3 {k m m' n : ℕ} {u v w x z : ℤ}
+  (hkm : k < m) (hmn : m < n) (hkm' : k < m') (hmn' : m' < n)
+  (hu : is_unit u) (hv : is_unit v) (hw : is_unit w) (hx : is_unit x) (hz : is_unit z)
+  (hp : p = C u * X ^ k + C v * X ^ m + C w * X ^ n)
+  (hq : q = C x * X ^ k + C v * X ^ m' + C z * X ^ n)
+  (h : p * p.mirror = q * q.mirror) :
+  q = p ∨ q = p.mirror :=
+begin
+  have hmul := congr_arg leading_coeff h,
+  rw [leading_coeff_mul, leading_coeff_mul, mirror_leading_coeff, mirror_leading_coeff, hp, hq,
+      trinomial_leading_coeff hkm hmn hw.ne_zero, trinomial_leading_coeff hkm' hmn' hz.ne_zero,
+      trinomial_trailing_coeff hkm hmn hu.ne_zero,
+      trinomial_trailing_coeff hkm' hmn' hx.ne_zero] at hmul,
+
+  have hadd := congr_arg (eval 1) h,
+  rw [eval_mul, eval_mul, mirror_eval_one, mirror_eval_one, ←sq, ←sq, hp, hq] at hadd,
+  simp only [eval_add, eval_C_mul, eval_pow, eval_X, one_pow, mul_one] at hadd,
+  rw [add_assoc, add_assoc, add_comm u, add_comm x, add_assoc, add_assoc] at hadd,
+  simp only [add_sq', add_assoc, add_right_inj, int.is_unit_sq, hu, hv, hw, hx, hz] at hadd,
+  rw [mul_assoc, hmul, ←mul_assoc, add_right_inj,
+      mul_right_inj' (show 2 * v ≠ 0, from mul_ne_zero two_ne_zero hv.ne_zero)] at hadd,
+
+  rcases (int.is_unit_add_is_unit_eq_is_unit_add_is_unit hw hu hz hx).mp hadd with
+    ⟨rfl, rfl⟩ | ⟨rfl, rfl⟩,
+  { exact irreducible_aux2 hkm hmn hkm' hmn' hu hv hw hp hq h },
+  { rw [←mirror_inj, trinomial_mirror hkm' hmn' hw.ne_zero hu.ne_zero] at hq,
+    rw [mul_comm q, ←q.mirror_mirror, q.mirror.mirror_mirror] at h,
+    rw [←mirror_inj, or_comm, ←mirror_eq_iff],
+    exact irreducible_aux2 hkm hmn (lt_add_of_pos_left k (tsub_pos_of_lt hmn'))
+      ((lt_tsub_iff_right).mp ((tsub_lt_tsub_iff_left_of_le hmn'.le).mpr hkm')) hu hv hw hp hq h },
+end
+
+lemma irreducible_of_coprime (hp : p.is_unit_trinomial)
+  (h : ∀ q : ℤ[X], q ∣ p → q ∣ p.mirror → is_unit q) :
```

Comment:
This is equivalent to `coprime` over a GCD domain like `ℤ[X]`, right?

Reply:
Not entirely sure, since `is_coprime` is defined in terms `a * x + b * y = 1`, but I can certainly add a version using `is_coprime`.